build/Test.ruleset

```diff
@@ -37,5 +37,10 @@
   <Rules AnalyzerId="xunit.analyzers" RuleNamespace="xunit.analyzers">
     <Rule Id="xUnit1026" Action="None" /> <!-- Theory methods must use all parameters -->
     <Rule Id="xUnit1004" Action="None" /> <!-- Test methods should not be skipped -->
+
+    <!-- Disabled these two rules until we pick up a compiler with the fix for orkaround https://github.com/dotnet/roslyn/issues/31740. -->
```

Comment:
>orkaround [](start = 79, length = 9)

typo: workaround

Did you understand why those XUnit diagnostics fire?

Reply:
I believe it was the same issue as attribute array nullability comparisons.

Reply:
Since I was not able to repro the XUnit issue, I'm not sure the compiler fix fixes it.

Reply:
This is a repro if you want to enlist ;-)

Reply:
@jcouv: FYI - Updated to a compiler with the fix for 31740, and this still repros. Also seems to cause a code-gen change leading to test failures when these errors are suppressed.